### PR TITLE
Feature 673 release guide metplus

### LINKED ISSUE #673: Internal: Create a METplus "Release Guide" describing how to build releases for the METplus components.
## Describe the Task ##
Please see [METplus meeting notes](https://docs.google.com/document/d/1nkAPFijlJq3X3vuOBVBPgVSzTn6IUMEAO_gnEaLskyg/edit#) from 10/27/2020 for more details. This task is to create a 3rd document for METplus in addition to the user and contributor's guides. This document will describe the process for creating a METplus release and consist of up to 8 chapters:
1. Introduction
2. METplus Release
3. MET Release
4. METdatadb Release
5. METcalcpy Release
6. METplotpy Release (or combine METcalcpy/METplotpy into a single chapter if that makes sense?)
7. METviewer Release
8. METexpress Release

The docs/Release_Guide directory will contain .rst files for each of the items listed above. It should also contain a subdirectory named "release_steps". Each top-level .rst file should be very sparse and consist basically of 3 bulleted lists:
- official release (e.g. vX.Y)
- bugfix release (e.g. vX.Y.Z)
- development release (e.g. vX.Y-betaN or vX.Y-rcN)

Each element in that list should source on .rst files that live in the "release_steps" sub-directory. It'd be cool if that content could be hidden by default, but click on it to expand it. So while we have separate lists for each component, they serve up common content from the "release_steps" directory.

Migrate content from:
Existing wiki page:
   https://github.com/dtcenter/METplus/wiki/Creating-a-new-METplus-release
Creating met-10.0_beta1 release:
   https://github.com/dtcenter/MET/issues/1523#issuecomment-714781834

### Time Estimate ###
4-8 hours for many different people.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*
May need to break this down, but not quite sure how to do so yet.

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2702691

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required: George (METplus), John HG (MET), Minna (METplotpy/METcalcpy), Tatiana (METviewer), Julie (docs infrastructure), Venita (METdatadb) and Molly (METexpress)
- [x] Select **scientist(s)** or **no scientist** required: Keith (review and advise)

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
Details for each component will live in the common METplus Release Guide. So all changes will be in the METplus repo.

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by jprestop**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
I reviewed the documentation in a temporary hosting location:
https://ral.ucar.edu/staff/jpresto/work/METplus_jpresto/Release_Guide/metplus.html

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Please review the files that changed and the way the are rendered here:
https://ral.ucar.edu/staff/jpresto/work/METplus_jpresto/Release_Guide/metplus.html
In your comments, please remind me to update the gh-pages if these changes are approved.  Thanks!

- [x] Do these changes include sufficient documentation and testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.
- [x] After the PR is approved, merge your changes (and update gh-pages). If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Release_Guide/release_steps/update_release_notes_bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+.. parsed-literal::
+
+    https://github.com/dtcenter/|projectRepo|/issues
+
+* Click on the Projects tab and select the project (under Repository) that
+  corresponds to the release you are creating.
+
+* Update the release-notes.rst file found in the User's Guide directory.
+
+* In creating a bugfix release, leave the "Version X.Y.0 release notes
```
Comment (JohnHalleyGotway) [inline docs/Release_Guide/release_steps/update_release_notes_bugfix.rst]:
```suggestion
* When creating a bugfix release, leave the "Version X.Y.0 release notes
```

File: docs/Release_Guide/release_steps/update_release_notes_development.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+.. parsed-literal::
+
+    https://github.com/dtcenter/|projectRepo|/issues
+
+* Click on the Projects tab and select the project (under Repository) that
+  corresponds to the release you are creating.
+
+* Update the release-notes.rst file found in the User's Guide directory.
+
+* If you are creating a beta1 release, remove the previous version's release
```
Comment (JohnHalleyGotway) [inline docs/Release_Guide/release_steps/update_release_notes_development.rst:16]: This same instruction is listed twice.

File: docs/Release_Guide/release_steps/update_release_notes_official.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ release. Open the following URL in a browser:
 
 * Update the release-notes.rst file found in the User's Guide directory.
 
-* If you are creating a beta1 release, remove the previous version's release
-  notes, i.e. for 3.0-beta1, remove all 2.Y notes and start the 3.0 section.
-
+* In creating the official release, combine the beta release sections
```
Comment (JohnHalleyGotway) [inline docs/Release_Guide/release_steps/update_release_notes_official.rst]:
```suggestion
* When creating the official release, combine the beta release sections
```

File: docs/Users_Guide/release-notes.rst [hunk 4]
```diff
@@ -1,95 +1,122 @@
...
-* Add stage to Travis to update or create data volumes when new sample data is available (`#633 <https://github.com/dtcenter/METplus/issues/633>`_)
-* Docker data caching (`#623 <https://github.com/dtcenter/METplus/issues/623>`_)
-* Tutorial testing on supported platforms (`#468 <https://github.com/dtcenter/METplus/issues/468>`_)
-* Add additional Branch support to the Travis CI pipeline (`#478 <https://github.com/dtcenter/METplus/issues/478>`_)
-* Change $DOCKER_WORK_DIR from /metplus to /root to be consistent with METplus tutorial (`#595 <https://github.com/dtcenter/METplus/issues/595>`_)
-* Add all use_cases to automated tests (eg Travis) (`#571 <https://github.com/dtcenter/METplus/issues/571>`_)
-* Add support to run METplus tests against multiple version of Python (`#483 <https://github.com/dtcenter/METplus/issues/483>`_)
+* New Wrappers:
+
+  * Met Tool Wrapper: PlotDataPlane/PlotDataPlane_grib1
```
Comment (JohnHalleyGotway) [inline docs/Users_Guide/release-notes.rst]:
```suggestion
  * MET Tool Wrapper: PlotDataPlane/PlotDataPlane_grib1
```
We should capitalize MET here and on several following lines.

**Review by JohnHalleyGotway — CHANGES_REQUESTED** [review]
Julie, overall looks great. There were 4 (I think) individual comments I made. Just a couple more edits needed.

**Review by JohnHalleyGotway — APPROVED** [review]
Looks excellent to me Julie! Thanks for working through these updates.
I made a couple minor edits, just on the number of stars above/below a couple of titles... making them the same width as the enclosed text. But no content changes.
I approve of these updates.

File: docs/Release_Guide/release_steps/update_release_notes_development.rst (hunk 2 above)
Comment (jprestop) [reply docs/Release_Guide/release_steps/update_release_notes_development.rst:16]: Thanks for catching that. I just fixed it up.

File: docs/Release_Guide/release_steps/update_release_notes_bugfix.rst (hunk 1 above)
Comment (jprestop) [reply docs/Release_Guide/release_steps/update_release_notes_bugfix.rst]: Done!

**Pull request merged**